### PR TITLE
feature(split-sct-log): small util for split sct log by nemesis

### PR DESCRIPTION
**Pull request by soyacz**
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by fruch — APPROVED** [review]
LGTM

**Review by fgelcer — DISMISSED** [review]
other than a small question/comment, LGTM

File: utils/split_sct_log.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    os.mkdir(input_dir / "parts")
+
+with open(input_dir / "sct.log", "r", encoding="utf-8") as sct_log:
+    idx = 0  # pylint: disable=invalid-name
+    line = sct_log.readline()
+    time = get_time(line)
+    part_file = open(input_dir / "parts" / f"{idx:03d}_{time}_start.log", "w",  # pylint: disable=consider-using-with
+                     encoding="utf-8")
+    part_file.write(line)
+    for line in sct_log.readlines():
```
[the remark this answers is not in the document]
Comment (soyacz) [reply utils/split_sct_log.py:31]: it does not make difference - since line has been written, we don't care.

File: utils/split_sct_log.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
```
Comment (enaydanov) [inline utils/split_sct_log.py:1]: You need to add license preamble and copyright string.

File: utils/split_sct_log.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+chmod +x utils/split_sct_log.py
+sudo ln -s <sct_repo_dir>/utils/split_sct_log.py /usr/bin/split_sct_log.py
+"""
+
+import os
+from pathlib import Path
+
+
+def get_time(line_content):
+    return "_".join(line_content.split(" ")[2:3]).split(",", maxsplit=1)[0]
```
Comment (enaydanov) [inline utils/split_sct_log.py:16]: Please, provide an example of a string parsed.

File: utils/split_sct_log.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+input_dir = Path(os.getcwd())
```
Comment (enaydanov) [inline utils/split_sct_log.py:19]: `Path.cwd()`

File: utils/split_sct_log.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+if not Path.exists(input_dir / "parts"):
+    os.mkdir(input_dir / "parts")
```
Comment (enaydanov) [inline utils/split_sct_log.py:22]: `Path(input_dir / "parts").mkdir(exist_ok=True)`

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: utils/split_sct_log.py (hunk 5 above)
Comment (soyacz) [reply utils/split_sct_log.py]: Thanks @enaydanov !

File: utils/split_sct_log.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2021 ScyllaDB
```
Comment (enaydanov) [inline utils/split_sct_log.py:13]: 2022 :)

Comment (soyacz) [reply utils/split_sct_log.py]: time goes by so fast... :smile:

**Review by enaydanov — APPROVED** [review]

**Review by fgelcer — APPROVED** [review]
LGTM

**Pull request merged**